vendor/engines/split_accounts/app/services/split_accounts/reports/querier.rb

```diff
@@ -0,0 +1,29 @@
+module SplitAccounts
+
+  module Reports
+
+    class Querier < ::Reports::Querier
+
+      def perform
+        super.map do |order_detail|
+          simulate_order_details(order_detail)
+        end.flatten
+      end
+
+      def simulate_order_details(order_detail)
+        if order_detail.account.is_a?(SplitAccounts::SplitAccount)
```

Comment:
My other thought is that if we do keep the `is_a?` check, that it gets pulled into `SplitOrderDetailSimulator`. Then this class becomes pretty trivial and the simulator is what you're actually overriding in Settings, not the Querier, since the query is always the same.